engine/api/src/main/scala/pl/touk/nussknacker/engine/api/process/ClassExtractionSettings.scala

```diff
@@ -5,27 +5,131 @@ import java.util.regex.Pattern
+      ClassPatternPredicate(Pattern.compile("java\\.lang\\.Void")),
+      ClassPatternPredicate(Pattern.compile("scala\\.Unit.*")),
+      ClassPatternPredicate(Pattern.compile("scala\\.runtime\\.BoxedUnit")),
+
+      // In case if there is some public method with flink's TypeInformation on serialization purpose
+      ClassPatternPredicate(Pattern.compile("org\\.apache\\.flink\\.api\\.common\\.typeinfo\\.TypeInformation")),
+      // We use this type only programmable
+      ClassPatternPredicate(Pattern.compile("pl\\.touk\\.nussknacker\\.engine\\.spel\\.SpelExpressionRepr")),
+      // In case if someone use it for kind of meta programming
+      ClassPatternPredicate(Pattern.compile("java\\.lang\\.Class")),
+      // In case if someone return function for lazy evaluation purpose
+      ClassPatternPredicate(Pattern.compile("java\\.util\\.function\\..*")),
+      ClassPatternPredicate(Pattern.compile("scala\\.Function.*")),
+      // java xml api is not easy to use without additional helpers, so we will skip these classes
+      ClassPatternPredicate(Pattern.compile("javax\\.xml\\..*")),
+      // Not sure why these below exclusions are TODO describe why they should be here or remove it
+      ClassPatternPredicate(Pattern.compile("dispatch\\..*")),
+      ClassPatternPredicate(Pattern.compile("cats\\..*"))
+    )
+
+  lazy val DefaultBlacklistedMembers: List[ClassMemberPredicate] = CommonBlacklistedMembers ++ AvroBlacklistedMembers
+
+  lazy val CommonBlacklistedMembers: List[ClassMemberPredicate] =
+    List(
+      // We want to hide all technical methods in every class, toString can be useful so we will leave it
+      AllMethodNamesPredicate(classOf[DumpCaseClass], Set(ToStringMethod))
+    )
+
+  lazy val AvroBlacklistedMembers: List[ClassMemberPredicate] =
     List(
       ClassMemberPatternPredicate(
-        Pattern.compile("org\\.apache\\.avro\\.generic\\.IndexedRecord"),
+        SuperClassPatternPredicate(Pattern.compile("org\\.apache\\.avro\\.generic\\.IndexedRecord")),
         Pattern.compile("(getSchema|compareTo|put)")),
       ClassMemberPatternPredicate(
-        Pattern.compile("org\\.apache\\.avro\\.specific\\.SpecificRecordBase"),
+        SuperClassPatternPredicate(Pattern.compile("org\\.apache\\.avro\\.specific\\.SpecificRecordBase")),
         Pattern.compile("(getConverion|getConversion|writeExternal|readExternal|toByteBuffer|set[A-Z].*)"))
     )
 
-}
\ No newline at end of file
+  lazy val DefaultWhitelistedMembers: List[ClassMemberPredicate] = WhitelistedUtilMembers ++ WhitelistedSerializableMembers ++ WhitelistedStdMembers
+
+  lazy val WhitelistedUtilMembers: List[ClassMemberPredicate] =
+    List(
+      // For numeric types, strings an collections, date types we want to see all useful methods
+      ClassMemberPatternPredicate(
+        SuperClassPatternPredicate(Pattern.compile("(java\\.lang\\.Number|java\\.util\\.Date|java\\.util\\.Calendar|java\\.util\\.concurrent\\.TimeUnit)")),
```

Comment:
We need this rule because these classes are inside `java.*` package and we have more general rule for classes inside `java.*`. It can be tricky now and it was a cause, why I've picked first matching inclusive rule at the beginning.... Maybe I should add this info in comment and move this general rule closer to this?

Reply:
Moved closer + added some comment

Reply:
thx